worker-core/src/main/java/com/hpe/caf/worker/core/WorkerCore.java

```diff
@@ -172,33 +175,33 @@ private void registerNewTaskImpl(final TaskInformation taskInformation, final by
                         + " (Assuming task {} is no longer active. The task message (message id: {}) will not be executed)",
                         tm.getTaskId(),
                         taskInformation.getInboundMessageId());
-                    executor.discardTask(tm, taskInformation);
+                    executor.discardTask(tm, coreTaskInformation);
                     return;
                 }
                 switch (jobStatus) {
                     case Active:
                     case Waiting:
                         if (isTaskIntendedForThisWorker(tm, taskInformation)) {
-                            executor.executeTask(tm, taskInformation, poison, headers, codec);
+                            executor.executeTask(tm, coreTaskInformation, poison, headers, codec);
                         } else {
-                            executor.handleDivertedTask(tm, taskInformation, poison, headers, codec, jobStatus);
+                            executor.handleDivertedTask(tm, coreTaskInformation, poison, headers, codec, jobStatus);
                         }
                         break;
                     case Paused:
                         if (isTaskIntendedForThisWorker(tm, taskInformation)) {
                             final String pausedQueue = workerQueue.getPausedQueue();
                             if (pausedQueue != null) {
-                                executor.pauseTask(tm, taskInformation, pausedQueue, headers);
+                                executor.pauseTask(tm, coreTaskInformation, pausedQueue, headers);
                             } else {
                                 LOG.debug(
                                     "Task {} is paused but the paused queue has not been set. "
                                     + "Task message (message id: {}) will be executed as normal",
                                     tm.getTaskId(),
                                     taskInformation.getInboundMessageId());
-                                executor.executeTask(tm, taskInformation, poison, headers, codec);
+                                executor.executeTask(tm, coreTaskInformation, poison, headers, codec);
                             }
                         } else {
-                            executor.handleDivertedTask(tm, taskInformation, poison, headers, codec, jobStatus);
+                            executor.handleDivertedTask(tm, coreTaskInformation, poison, headers, codec, jobStatus);
                         }
                         break;
                     default:
```

Comment:
`executor.discardTask` not passing wrapped task information.

Reply:
Fixed in 4171002.

Reply:
`discardTask` required only the `inboundMessageId` which was available from the `taskInformation` too. Isn't wrapping required only when we need access to `taskData` ? Or is this being done to maintain consistency and not maintain two versions of the similar object?

Reply:
I figured that might be why you left it out and you're right that with the current implementation that it wouldn't cause an issue.  But the code overall is easier to reason about if the `WorkerExecutor` is only ever passed the wrapped task information.  You always know it has to be unwrapped in the callback.  (Looking at the changes now I suspect that the `WeakHashMap` solution would probably have been less changes - but I also think this solution is fine).